conan/tools/premake/premake.py

```diff
@@ -1,40 +1,91 @@
+        # Path to the root (premake5) lua file
+        self.luafile = (Path(self._conanfile.source_folder) / "premake5.lua").as_posix()
         # (key value pairs. Will translate to "--{key}={value}")
-        self.arguments = {}             # https://premake.github.io/docs/Command-Line-Arguments/
+        self.arguments = {}  # https://premake.github.io/docs/Command-Line-Arguments/
+
+        arch = str(self._conanfile.settings.arch)
+        if arch not in CONAN_TO_PREMAKE_ARCH:
+            raise ConanException(f"Premake does not support {arch} architecture.")
+        self.arguments["arch"] = CONAN_TO_PREMAKE_ARCH[arch]
 
         if "msvc" in self._conanfile.settings.compiler:
-            msvc_version = PREMAKE_VS_VERSION.get(str(self._conanfile.settings.compiler.version))
-            self.action = f'vs{msvc_version}'
+            msvc_version = PREMAKE_VS_VERSION.get(
+                str(self._conanfile.settings.compiler.version)
+            )
+            self.action = f"vs{msvc_version}"
         else:
-            self.action = 'gmake2'
+            self.action = "gmake" # New generator (old gmakelegacy is deprecated)
 
     @staticmethod
     def _expand_args(args):
-        return ' '.join([f'--{key}={value}' for key, value in args.items()])
+        return " ".join([f"--{key}={value}" for key, value in args.items()])
 
     def configure(self):
+        premake_conan_toolchain = Path(self._conanfile.generators_folder) / PremakeToolchain.filename
+        if not premake_conan_toolchain.exists():
+            raise ConanException(
+                f"Premake toolchain file does not exist: {premake_conan_toolchain}\nUse PremakeToolchain to generate it."
+            )
+        content = Template(self._premake_file_template).render(
+            premake_conan_toolchain=premake_conan_toolchain, luafile=self.luafile
+        )
+
+        conan_luafile = Path(self._conanfile.build_folder) / self.filename
+        save(self._conanfile, conan_luafile, content)
+
         premake_options = dict()
-        premake_options["file"] = self.luafile
+        premake_options["file"] = conan_luafile
 
-        premake_command = f'premake5 {self._expand_args(premake_options)} {self.action} ' \
-                          f'{self._expand_args(self.arguments)}'
+        premake_command = (
+            f"premake5 {self._expand_args(premake_options)} {self.action} "
+            f"{self._expand_args(self.arguments)}"
+        )
         self._conanfile.run(premake_command)
+
+
+    def build(self, workspace, targets=None):
+        if self.action.startswith("vs"):
+            msbuild = MSBuild(self._conanfile)
+            msbuild.build(sln=f"{workspace}.sln", targets=targets)
+        else:
+            build_type = str(self._conanfile.settings.build_type)
+            targets = "all" if targets is None else " ".join(targets)
+            self._conanfile.run(f"make config={build_type.lower()} {targets} -j")
```

Comment:
Do not hardcode ``-j`` (I assume that will use the maximum cores?); but use the ``conf`` that models the cores for parallel building

Reply:
Good idea, I have just updated it